### PR TITLE
Create action

### PR DESCRIPTION
**Pull request by ben-harvey**
This change creates a composite action to run the `scan-github` program in https://github.com/Enterprise-CMCS/mac-fc-dso-metrics. As a convenience to the caller, It adds an input for an OIDC role with permissions to assume the DSO Metrics cross-account role.

The composite action calls a local Typescript action that fetches a release binary from https://github.com/Enterprise-CMCS/mac-fc-scan-github-releases that matches the specified version constraint and the runner type.

Testing: 
- I ran the action in the ProfessorMAC repo and confirmed in the AWS console that it reported DSO events

### CODE REVIEW

File: .github/actions/scan-github/action.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+description: 'Uses the GitHub API to create events from past workflow runs, and sends event data to the MACBIS DevSecOps Metrics API'
+inputs:
+  version:
+    description: The version constraint for the scan-github program in semantic version syntax. Defaults to the latest version (we recommend pinning to a specific version or range)
+    default: "*"
+  github-access-token:
+    description: A GitHub token with repo scope for the target repo
+    default: ${{ github.token }}
+  config:
+    description: A yaml file with a valid configuration for the scan-github tool. See Enterprise-CMCS/mac-fc-dso-metrics/cmd/scan-github/README.md for the syntax of the confuration
```
Comment (hundt-corbalt) [inline .github/actions/scan-github/action.yml:11]: typo: "confuration"

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+name: 'Scan GitHub for DevSecOps (DSO) Events'
+description: 'Uses the GitHub Actions API to scan for event data to send the MACBIS DSO Metrics API'
+inputs:
+  scan-github-version:
+    description: 'The version constraint for the Enterprise-CMCS/mac-fc-dso-metrics/cmd/scan-github program in semantic version syntax. Defaults to the latest version (we recommend pinning to a specific version or range)'
+    required: false
```
Comment (hundt-corbalt) [inline action.yml:6]: how does it work that this is not required but has no default? does that mean it is `null` if the user doesn't specify it? will that correctly trigger the default behavior in the underlying action?

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
+import { spawnSync } from "child_process";
```
Comment (hundt-corbalt) [inline src/index.ts:1]: I guess the next step is to release this script as an NPM module so that we can use it in multiple git repos :)

**Review by hundt-corbalt — COMMENTED** [review]
nice!

File: .github/actions/scan-github/action.yml (hunk 1 above)
Comment (ben-harvey) [reply .github/actions/scan-github/action.yml]: :thumbsup:

File: action.yml (hunk 2 above)
Comment (ben-harvey) [reply action.yml:6]: Ah, the default is set by the local Typescript action but not here in the composite action, which is confusing. I'll set it here too

File: src/index.ts (hunk 3 above)
Comment (ben-harvey) [reply src/index.ts:1]: That's not a bad idea :)

**Review by hundt-corbalt — APPROVED** [review]
LGTM!

**Pull request merged**